contracts/DatasourceDelegate/examples/whitelist/WhitelistDataSource.sol

```diff
@@ -0,0 +1,158 @@
+    _;
+  }
+
+  /**
+    @param _jbProjects the JBProjects contracts deployment
+    @param _projectId the id of the project (the holder of the corresponding NFT has access to add/remove address)
+  */
+  constructor(IJBProjects _jbProjects, uint256 _projectId) {
+    jbProjects = _jbProjects;
+    projectId = _projectId;
+  }
+
+  //@inheritdocs IJBFundingCycleDataSource
+  function payParams(JBPayParamsData calldata _data)
+    external
+    view
+    override
+    returns (
+      uint256 weight,
+      string memory memo,
+      IJBPayDelegate delegate
+    )
+  {
+    // Revert if beneficiary is non whitelisted
+    if (!isWhitelisted[_data.beneficiary]) revert WhitelistDataSource_BeneficiaryNotWhitelisted();
+
+    // If disabling non-whitelisted payers too, uncomment the following:
+    // if (!isWhitelisted[_data.payer]) revert WhitelistDataSource_BeneficiaryNotWhitelisted();
+
+    return (_data.weight, _data.memo, IJBPayDelegate(address(0)));
+  }
+
+  /**
+    @notice  Add an array of addresses to the whitelist
+    @dev     the previous whitelist status is not read, for gas optimisation
+    @param   addresses the addresses to whitelist
+  */
+  function addToWhitelist(address[] calldata addresses) external onlyProjectOwner {
+    uint256 numberOfAddresses = addresses.length; // Push to stack
+    for (uint256 i; i < numberOfAddresses; ) {
+      isWhitelisted[addresses[i]] = true;
+      unchecked {
+        ++i;
+      }
+    }
+    emit AddToWhitelist(numberOfAddresses); // Not passing the whole array for gas reason
+  }
+
+  /**
+    @notice  Add a single address to the whitelist
+    @param   _address the address to whitelist
+  */
+  function addToWhitelist(address _address) external onlyProjectOwner {
+    isWhitelisted[_address] = true;
+    emit AddToWhitelist(1);
+  }
+
+  /**
+    @notice  Remove an array of addresses from the whitelist
+    @dev     the previous whitelist status is not read, for gas optimisation
+    @param   addresses the addresses to remove from the whitelist
+  */
+  function removeFromWhitelist(address[] calldata addresses) external onlyProjectOwner {
+    uint256 numberOfAddresses = addresses.length; // Push to stack
+    for (uint256 i; i < numberOfAddresses; ) {
+      delete isWhitelisted[addresses[i]];
+      unchecked {
+        ++i;
+      }
+    }
+    emit RemoveFromWhitelist(numberOfAddresses);
+  }
+
+  /**
+    @notice  Remove a single address from the whitelist
+    @dev     the previous whitelist status is not read, for gas optimisation
+    @param   _address the addresses to remove from the whitelist
+  */
+  function removeFromWhitelist(address _address) external onlyProjectOwner {
+    delete isWhitelisted[_address];
+    emit RemoveFromWhitelist(1);
+  }
+
+  function supportsInterface(bytes4 _interfaceId) external pure override returns (bool) {
+    return _interfaceId == type(IJBFundingCycleDataSource).interfaceId;
```

Comment:
Make sure it also reflects implementing the ERC165 standard itself by returning true for the ERC165 interfaceId ([Example OZ](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/introspection/ERC165.sol))

Reply:
Good catch, I forgot to `|| super`